lib/elixir/lib/process.ex

```diff
@@ -399,8 +398,8 @@ defmodule Process do
 
   Inlined by the compiler.
   """
-  @spec demonitor(reference) :: true
   @spec demonitor(reference, options :: [:flush | :info]) :: boolean
+  @spec demonitor(reference, []) :: true
```

Comment:
This change introduces an overlapping domain which isn't handle by dialyzer. Therefore the addition would be ignored by dialyzer if the spec was ever used - this function is inlined by the compiler so `:erlang.demonitor/1,2` spec is used.

The domains wouldn't overlap if it was `[:flush | :info, ...]`. However having both for documentation is not clear.